controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -134,6 +140,9 @@ func (r *StorageConsumerReconciler) initReconciler(request reconcile.Request) {
 }
 
 func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error) {
+	if err := r.reconcileOnboardingSecret(); err != nil {
+		return reconcile.Result{}, err
+	}
```

Comment:
I would expect to see a clear separation between
`reconcileEnabledPhases` and `reconcileNonEnabledPhases`

Reply:
only did partially for upcoming changes (moving from storagerequest to storageclient) to rearrange the reconcile steps.

a doubt, how to handle b/n storageconsumer existence & deletion w/ enable & notenable then?
```
if enable {
 if time-stamp-not-zero { ... }
 else { ... }
} else {
 if time-stamp-not-zero { ... }
 else { ... }
}
```